lib/utils/utilities.js

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env node
+
+var shelljs = require('shelljs');
+var verbose = undefined;
+var fs      = require('fs');
+var os      = require("os");
+var util    = require('util');
+var path    = require("path-extra");
+var logger  = require('cordova-common').CordovaLogger.get();
+
+var HEADING_LINE_PATTERN = /List of devices/m;
+var DEVICE_ROW_PATTERN   = /(emulator|device|host)/m;
+
+function isWindows () {
+    return /^win/.test(os.platform());
+}
+
+function countAndroidDevices() {
+    var listCommand = "adb devices";
+
+    logger.info("running:");
+    logger.info("    " + listCommand);
+
+    var numDevices = 0;
+    var result = shelljs.exec(listCommand, {silent: false, async: false});
+    result.output.split('\n').forEach(function (line) {
+        if (!HEADING_LINE_PATTERN.test(line) && DEVICE_ROW_PATTERN.test(line)) {
+            numDevices += 1;
+        }
+    });
+    return numDevices;
+}
+
+function secToMin (seconds) {
+    return Math.ceil(seconds / 60);
+}
+
+function getSimulatorsFolder() {
+    var simulatorsFolderPath = path.join(path.homedir(), "Library", "Developer", "CoreSimulator", "Devices");
+    return simulatorsFolderPath;
+}
+
+function getSimId() {
+    var findSimCommand = "cordova run --list --emulator | grep ^iPhone | tail -n1";
+
+    logger.info("running:");
+    logger.info("    " + findSimCommand);
+
+    var findSimResult = shelljs.exec(findSimCommand, {silent: true, async: false});
+
+    if (findSimResult.code > 0) {
+        logger.error("Failed to find simulator we deployed to");
+        return;
+    }
+
+    var split = findSimResult.output.split(", ");
+
+    // Format of the output is "iPhone-6s-Plus, 9.1"
+    // Extract the device name and the version number
+    var device = split[0].replace(/-/g, " ").trim();
+    var version = split[1].trim();
+
+    // Next, figure out the ID of the simulator we found
+    var instrCommand = "instruments -s devices | grep ^iPhone";
+    logger.info("running:");
+    logger.info("    " + instrCommand);
+
+    var instrResult = shelljs.exec(instrCommand, {silent: true, async: false});
+
+    if (instrResult.code > 0) {
+        logger.error("Failed to get the list of simulators");
+        return;
+    }
+
+    // This matches <device> (<version>) [<simulator-id>]
+    var simIdRegex = /^([a-zA-Z\d ]+) \(([\d.]+)\) \[([a-zA-Z\d\-]*)\]$/;
+
+    var simId = null;
+    var lines = instrResult.output.split(/\n/);
+    lines.forEach(function(line) {
+        var simIdMatch = simIdRegex.exec(line);
+        if (simIdMatch && simIdMatch.length === 4 && simIdMatch[1] === device && simIdMatch[2] === version) {
+            simId = encodeURIComponent(simIdMatch[3]);
+        }
+    });
+
+    return simId;
+}
```

Comment:
So, running this code, I face an issue where the simId returned is null.
Here's the output of various bash commands executed here:
- 
```
nstruments -s devices | grep ^iPhone
iPhone 4s (9.3) [93EC91E3-2679-49DC-AC6B-CAB279E462AF] (Simulator)
iPhone 5 (9.3) [830AC555-1986-4B48-82C7-898849A50541] (Simulator)
iPhone 5s (9.3) [17726EFF-F3B9-435E-8882-CC6ED589C213] (Simulator)
iPhone 6 (9.3) [3BDE39CD-7998-4959-B2D8-7C08C33B20B1] (Simulator)
iPhone 6 Plus (9.3) [7235ADB5-8730-4D0A-821F-5469D1ED47D7] (Simulator)
iPhone 6s (9.3) [32745196-42F6-4927-ADA0-16BACE15FC99] (Simulator)
iPhone 6s (9.3) + Apple Watch - 38mm (2.2) [6A868DF2-F36A-4AEE-9EC1-B956375934B9] (Simulator)
iPhone 6s Plus (9.3) [A949CB17-350F-4B28-AD64-425339ACEDBF] (Simulator)
iPhone 6s Plus (9.3) + Apple Watch - 42mm (2.2) [E0C9BC62-83AA-45C5-B411-FBCB557DE32B] (Simulator)
```